src/hotspot/share/classfile/stackMapTableFormat.hpp

```diff
@@ -478,7 +478,7 @@ class chop_frame : public stack_map_frame {
   }
 
   static u1 chops_to_frame_type(int chop) {
-    return 251 - chop;
+    return (u1)(251 - chop);
```

Comment:
This looks dangerous. If the range is alway within `u1`, it should use `checked_cast<u1>`, and you should also assert `chop <= 251`.

Reply:
I agree, this one I should have used checked_cast<u1>.

Reply:
I also wondered about the range check but then I looked at the caller. No need to check it it in both places.

Reply:
Since the caller isn't easy to find (via grep), I think the checked_cast<> doesn't hurt anything.